### DISCUSSION TITLE
Unbound config systemctl warning

### DISCUSSION THREAD

**Discussion by jsapede** [community-scripts/ProxmoxVE#686] [General]
installed this morning and got the following warning messages with `systemctl status unbound`

```
● unbound.service - Unbound DNS server
     Loaded: loaded (/lib/systemd/system/unbound.service; enabled; preset: enabled)
     Active: active (running) since Thu 2024-12-05 08:38:38 CET; 1min 58s ago
       Docs: man:unbound(8)
   Main PID: 3723 (unbound)
      Tasks: 1 (limit: 9275)
     Memory: 7.8M
        CPU: 92ms
     CGroup: /system.slice/unbound.service
             └─3723 /usr/sbin/unbound -d -p

Dec 05 08:38:37 unbound systemd[1]: Starting unbound.service - Unbound DNS server...
Dec 05 08:38:37 unbound unbound[3723]: [1733384317] unbound[3723:0] warning: so-rcvbuf 1048576 was not granted. Got 425984. To fix: start with root permissions>
Dec 05 08:38:38 unbound systemd[1]: Started unbound.service - Unbound DNS server.
```

**Comment by havardthom** [comment]
@wimb0

**Comment by wimb0** [comment]
I will look in to this.
I'm seeing the same message, but Unbound seems to work fine.

Seems like setting sysctl settings inside the LXC is not working. Alternative is running the LXC privileged (not a good idea)...

```
root@unbound:/etc/unbound/unbound.conf.d# sysctl -w net.core.rmem_max=1048576
sysctl: setting key "net.core.rmem_max": Invalid argument
```

**Comment by wimb0** [comment]
I've checked the PiHole LXC with Unbound and that has the same issue btw.

Unbound has a config option for `so-rcvbuf:` , default is 1M, I will test with a smaller value.

**Reply by havardthom** [reply]
I think setting `so-rcvbuf: 1048576k` explicitly instead of using `1m` will fix the issue, ref. https://github.com/trinib/AdGuard-WireGuard-Unbound-DNScrypt/issues/32#issue-1235970778

**Reply by wimb0** [reply]
I've tested that, and that does not solve the issue.
The default max so-rcvbuf of Proxmox is 425984, which is smaller than 1048576.
To remove the error, we could set is to 425984. or change it on the Proxmox host to 1048576
